integration-tests/test/sync-verifier.spec.ts

```diff
@@ -0,0 +1,149 @@
+import chai, { expect } from 'chai'
+import { Wallet, BigNumber, Contract, ContractFactory, providers } from 'ethers'
+import { ethers } from 'hardhat'
+import { injectL2Context } from '@eth-optimism/core-utils'
+
+import { sleep, l2Provider } from './shared/utils'
+import { OptimismEnv } from './shared/env'
+import { DockerComposeNetwork } from './shared/docker-compose'
+
+describe('Syncing a verifier', () => {
+  let env: OptimismEnv
+  let wallet: Wallet
+  let verifier: DockerComposeNetwork
+
+  const provider = injectL2Context(l2Provider)
+
+  /** Helper functions **/
+
+  const waitForBatchSubmission = async (totalElementsBefore: BigNumber) => {
+    // Wait for batch submission to happen by watching the CTC
+    let totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    while (totalElementsBefore.eq(totalElementsAfter)) {
+      await sleep(500)
+      console.log(
```

Comment:
note: i'll remove all these `console.log`s if we decide to merge the dummy tx test first; leaving these here in case someone else wants to pull and debug